conductor/lib/uploader.py

```diff
@@ -467,24 +472,17 @@ def upload_status_text(self):
 
         return formatted_text
 
-
     def print_status(self):
         logger.debug('starting print_status thread')
         update_interval = 3
 
-        def sleep():
-            time.sleep(update_interval)
-
-        while True:
-            if self.working:
-                try:
-                    logger.info(self.manager.worker_queue_status_text())
-                    logger.info(self.upload_status_text())
-                except Exception, e:
-                    print e
-                    print traceback.format_exc()
-                    # pass
-            sleep()
+        while self.working:
+            try:
+                logger.info(self.manager.worker_queue_status_text())
+                logger.info(self.upload_status_text())
+            except Exception:
```

Comment:
`except:` would also work as a catch-all.

Reply:
I've been going back and forth on this recently (due to recent revelations that I'm somewhat embarrassed about).  And coincidentally, this module makes for the perfect example!
As much as I have loved to catch all exceptions by simply not listing any specific types (when deemed appropriate), I didn't realize there was a subtle difference in doing so vs explicitly catching the Exception class.  See https://stackoverflow.com/questions/730764/how-to-properly-ignore-exceptions

Usually it doesn't matter if we "catch all" exception, but because this module's application (the uploader), is a long running (daemon) process, that can only be stopped via killing it (ctrl-c, sigint, sigterm, etc), we have to be thoughtful to not disrupt any logic in the uploader that is expecting to encounter ctrl-c (KeyboardInterrupt). 
Here's a great example of how you can unintentionally make trouble :  https://fman.io/blog/an-inescapable-while-loop-in-python/

side note: here's an interesting diagram/hierarchy of pythons exception classes (as well as general good reading):  https://airbrake.io/blog/python-exception-handling/class-hierarchy